src/Workspaces/Core/Portable/Execution/Checksum_Factory.cs

```diff
@@ -24,6 +24,18 @@ public static Checksum Create(Stream stream)
             }
         }
 
+        public static Checksum Create(string kind, Checksum checksum)
+        {
+            using (var stream = SerializableBytes.CreateWritableStream())
```

Comment:
consider:  This may not be a meaningful optimization, but would it make sense to only re-compute the checksum if ```checksum``` represents an "empty" checksum (from an empty/null input)?